packages/react-form-renderer/src/components/form-renderer.js

```diff
@@ -91,11 +97,11 @@ const FormRenderer = ({
               onReset,
               getState,
               valid,
-              clearedValue,
+              clearedValue: typeof schemaClearedValue !== 'undefined' ? schemaClearedValue : clearedValue,
```

Comment:
Shouldn't it check if the property exists? What if user wants to use `undefined` as the clearedValue because the renderer component is using something else?

Reply:
undefined is set by default

Reply:
```jsx
schema = {
  ....
  clearedValue: undefined,
  fields: ....
}

<FormRenderer ... clearedValue={null} />
```

Now, we are breaking the priority rule. (Yeah, the use case does not make much sense, but everything can happen.)

Reply:
That brings up the question again. Should we just move these props exclusively to the schema? And remove them from renderer props?

Reply:
These props are more implementation details stuff, so I would rather keep them outside schemas. At least some of them - having subscription defined in the schema does not make much sense to me, as it would be in each form the same. And also we should define some kind of boundary, what should be in the schema and what not. For example, we could also define onSubmit, onCancel in the schema, but that would mean a lot of function passing... so it's a tough question. :D

Reply:
I think that initial values, cleared value, and clearOnUmnout should be in the schema and the rest should be props. I would like to avoid having duplications though.

Reply:
clearedValue and clearOnUnmount; I agree. clearedValue is basically useful mostly for edit forms, so it makes sense to have it only in the schema. clearOnUnmnout: I haven't seen any use case yet, so, who knows :D ... however, I see an issue with initialValues as they are loaded outside the form, so that mean you will do a lot of:

```jsx
<Renderer
   ...
   schema={schema(initialValues)}
/>
```

Reply:
You are right